pkg/cmd/browse/browse.go

```diff
@@ -275,7 +277,7 @@ func parseFile(opts BrowseOptions, f string) (p string, start int, end int, err
 	}
 
 	p = filepath.ToSlash(parts[0])
-	if !path.IsAbs(p) {
+	if !path.IsAbs(p) && reflect.TypeOf(opts.GitClient) != reflect.TypeOf(&remoteGitClient{}) && os.Getenv("GH_REPO") == "" {
```

Comment:
While this code does work to check if the user has supplied the `--repo` flag or the `GH_REPO` environment variable, I dislike that the intention behind the checks are not clear and I dislike the relying on `reflect` package to achieve it. 

What about moving this check up into the `RunE` section above and passing down the result of the check in the `opts`?

```go
RunE: func(cmd *cobra.Command, args []string) error {
...
    if cmd.Flags().Changed("repo") || os.Getenv("GH_REPO") != "" {
        opts.HasRepoOverride = true
    }
...
}
```

Then we can change this logic to:

```suggestion
	if !path.IsAbs(p) && !opts.HasRepoOverride {
```

Let me know your thoughts, or if that suggestion is not clear.

Reply:
@samcoe Thanks for your feedback! Your suggestion is clear and the code looks more reasonable. Yours is preferred imo.